packages/apex-lsp-vscode-extension/package.json

```diff
@@ -626,21 +626,143 @@
     }
   },
   "scripts": {
-    "precompile": "shx cp -R ../../node_modules/@salesforce/apex-tmlanguage/grammars . && shx mkdir -p out && shx cp package.nls*.json language-configuration.json out/",
-    "compile": "tsc --build",
-    "postcompile": "shx mkdir -p out/resources && shx cp ../apex-parser-ast/out/resources/StandardApexLibrary.zip out/resources/",
-    "bundle": "tsx esbuild.config.ts",
-    "copy-worker-files": "node scripts/postbundle.js",
-    "package": "cd dist && vsce package --out ../ --no-dependencies",
+    "precompile": "wireit",
+    "compile": "wireit",
+    "bundle": "wireit",
+    "copy-worker-files": "wireit",
+    "package": "wireit",
     "rebuild": "npm run clean && npm run build",
-    "copy:files": "shx cp -R ../../node_modules/@salesforce/apex-tmlanguage/grammars .",
-    "test": "jest --passWithNoTests",
-    "test:coverage": "jest --coverage --coverageDirectory=./coverage --passWithNoTests",
-    "clean": "rimraf out dist extension .turbo coverage grammars tsconfig.tsbuildinfo server-bundle",
+    "copy:files": "wireit",
+    "test": "wireit",
+    "test:coverage": "wireit",
+    "clean": "rimraf out dist extension .wireit coverage grammars tsconfig.tsbuildinfo server-bundle",
     "clean:all": "npm run clean && rimraf node_modules",
     "clean:coverage": "rimraf coverage",
-    "lint": "eslint src test",
-    "lint:fix": "eslint src test --fix"
+    "lint": "wireit",
+    "lint:fix": "wireit"
+  },
+  "wireit": {
+    "precompile": {
+      "command": "shx cp -R ../../node_modules/@salesforce/apex-tmlanguage/grammars . && shx mkdir -p out && shx cp package.nls*.json language-configuration.json out/",
+      "output": [
+        "grammars/**"
+      ],
+      "files": [
+        "package.nls*.json",
+        "language-configuration.json"
+      ]
+    },
+    "compile": {
+      "command": "tsc --build",
+      "dependencies": [
+        "precompile",
+        "../apex-parser-ast:precompile"
+      ],
+      "output": [
+        "out/**/*.js",
+        "out/**/*.d.ts",
+        "out/.tsbuildinfo"
+      ],
+      "files": [
+        "src/**/*.ts",
+        "tsconfig*.json"
+      ]
+    },
+    "bundle": {
+      "command": "tsx esbuild.config.ts",
+      "dependencies": [
+        "compile"
+      ],
+      "output": [
+        "extension/**",
+        "dist/**",
+        "server-bundle/**",
+        "dist/package.json"
+      ],
+      "files": [
+        "src/**/*.ts",
+        "esbuild.config.ts",
+        "out/extension.js"
+      ]
+    },
+    "copy-worker-files": {
+      "command": "node scripts/postbundle.js",
+      "dependencies": [
+        "bundle",
+        "../apex-ls:bundle"
+      ],
+      "output": [
+        "dist/**"
+      ],
+      "files": [
+        "dist/package.json"
+      ]
+    },
+    "package": {
```

Comment:
is it possible to simplify this, or take the parts to other idempotent wireit steps so that the command is less complex?